src/js/scheduler.ts

```diff
@@ -38,6 +43,36 @@ function installPostMessageHandler() {
 
 installPostMessageHandler();
 
+function ensureSharedChannel() {
+  if (sharedChannel) return;
+  if (RUNTIME_ENV.IN_SAFARI || RUNTIME_ENV.IN_DENO || RUNTIME_ENV.IN_NODE)
+    return;
+  if (typeof globalThis.MessageChannel !== "function") return;
+
+  sharedChannel = new MessageChannel();
+
+  sharedChannel.port1.onmessage = () => {
+    // Drain the queue: process as many scheduled tasks as are currently queued.
+    while (queue.length) {
+      const handle = queue.shift()!;
+      const task = tasks[handle];
+      if (!task) continue;
+      try {
+        task();
+      } finally {
+        delete tasks[handle];
+      }
+    }
+  };
+  if (typeof sharedChannel.port1.start === "function") {
+    try {
+      sharedChannel.port1.start();
```

Comment:
The explicit `port1.start(`) call is technically redundant since assigning `onmessage` already starts the port automatically (per the HTML spec).

It’s kept mainly for clarity and to guard against edge or non-standard runtimes.
There’s no performance impact since it’s called once during initialization.

Happy to remove it if you think it’s unnecessary!